app/forms/journeys/additional_payments_for_teaching/eligible_itt_subject_form.rb

```diff
@@ -28,16 +28,11 @@ def chemistry_or_physics_available?
       def save
         return false unless valid?
 
-        # FIXME RL: Once this method writes to the journey session answers we
-        # update the initializer in
-        # AdditionalPaymentsForTeaching::QualificationDetailsForm
-        claim.assign_attributes(
-          eligibility_attributes: {eligible_itt_subject: eligible_itt_subject}
+        journey_session.answers.assign_attributes(
```

Comment:
This will need to set `teaching_subject_now` to `nil` if `eligible_itt_subject` has changed but not if `teaching_subject_now` answer came from `dqt` 🙃. See `Policies::EarlyCareerPayments::Eligibility#reset_dependent_answers` for how this reseting stuff currently works.
To make it more fun as we're part way through the migration we'll want to reset `teaching_subject_now` on both the claim and the journey_session. An example of all this is `app/forms/journeys/additional_payments_for_teaching/qualification_form.rb` in this pr https://github.com/DFE-Digital/claim-additional-payments-for-teaching/pull/2803/files

Once we've finished the migration it's probably best to delete all the reset dependent answers callbacks and just have the appropriate forms handle `nil`ing out dependent answers